src/appleseed.shaders/src/maya/as_maya_asGlass.osl

```diff
@@ -0,0 +1,263 @@
+    ]],
+    normal in_normalCamera = N
+    [[
+        string maya_attribute_name = "normalCamera",
+        string maya_attribute_type = "vector",
+        string maya_attribute_short_name = "n",
+        string label = "Bump Normal",
+        string page = "Bump Parameters"
+    ]],
+    int in_matteOpacityMode = 2
+    [[
+        string maya_attribute_name = "matteOpacityMode",
+        string maya_attribute_type = "enum",
+        string label = "Matte Opacity Mode",
+        string widget = "mapper",
+        string options = "Black Hole:0|Solid Matte:1|Opacity Gain:2",
+        string page = "Matte Opacity Parameters"
+    ]],
+    float in_matteOpacity = 1.0
+    [[
+        string maya_attribute_name = "matteOpacity",
+        string maya_attribute_type = "float",
+        string label = "Matte Opacity",
+        string page = "Matte Opacity Parameters"
+    ]],
+    int in_maximumRayDepth = 4
+    [[
+        int min = 0,
+        int max = 32,
+        int softmin = 0,
+        int softmax = 8,
+        string label = "Ray Depth",
+        string page = "Advanced Parameters"
+    ]],
+
+    output closure color out_outColor = 0
+    [[
+        string maya_attribute_name = "outColor",
+        string maya_attribute_type = "color",
+        string maya_attribute_short_name = "oc",
+        string widget = "null"
+    ]],
+    output closure color out_outTransparency = 0
+    [[
+        string maya_attribute_name = "outTransparency",
+        string maya_attribute_type = "color",
+        string maya_attribute_short_name = "ot",
+        string widget = "null"
+    ]],
+    output closure color out_outMatteOpacity = 0
+    [[
+        string maya_attribute_name = "outMatteOpacity",
+        string maya_attribute_type = "color",
+        string maya_attribute_short_name = "om",
+        string widget = "null"
+    ]]
+)
+{
+    int ray_depth = 0;
+    int status = getattribute("path:ray_depth", ray_depth);
+
+    if (status && ray_depth <= in_maximumRayDepth)
+    {
+        normal Nn = normalize(in_normalCamera);
+        vector tangent;
+
+        if (in_anisotropyAmount)
+        {
+            if (isconnected(in_anisotropyVectorMap))
+            {
+                vector vector_map = normalize(
+                    (vector) in_anisotropyVectorMap * 2.0 - 1.0);
+
+                tangent = normalize(
+                    vector_map[0] * Tn +
+                    vector_map[1] * Bn +
+                    vector_map[2] * Nn);
+            }
+            else
+            {
+                tangent = Tn;
+            }
+
+            if (in_anisotropyAngle)
+            {
+                tangent = rotate(
+                    tangent,
+                    in_anisotropyAngle * M_2PI,
+                    point(0),
+                    point(in_normalCamera));
+            }
+        }
+
+        string distribution = (in_distribution) ? "ggx" : "beckmann";
+
+        out_outColor = in_transmittanceAmount *
```

Comment:
in_transmittanceAmount should go inside the closure, multiplied with in_surfaceTransmittance.
In the BSDF there's a special treatment for surfacet transmittance.